packages/universal-test-runner/src/readProtocol.ts

```diff
@@ -39,20 +47,12 @@ function readVersion(input: string | undefined): string {
   return input ?? DEFAULT_VERSION
 }
 
-function readTestsToRun(input: string | undefined): TestCase[] {
-  const TEST_CASE_SEPARATOR = '|'
-  const TEST_LOCATION_SEPARATOR = '#'
+function readTestsToRun(input: string | undefined): string | undefined {
+  return input || undefined
+}
 
-  return (
```

Comment:
Moved to the function where we invoke the adapter, since we now have multiple possible input sources for testsToRun that get passed to the adapter. The readProtocol has just become a bare-bones "read from process.env" function where we can take care of an immediate input validation and warnings and make sure things are passed (unprocessed) to the adapter running function.